### DISCUSSION TITLE
Feature request: sort functions within a file

### DISCUSSION THREAD

**Discussion by rodrigocfd** [golang/vscode-go#1289] [Discussions]
Just a simple idea: sort the functions (and methods) alphabetically in a Go file.

I found myself doing this manually quite often, it would be very handy to do this automatically.

**Comment by hyangah** [comment]
Personally I'd prefer the code follows the logic rather than lexical order, and let the Outline view or documentation (go doc) do sorting if necessary. But that's an interesting idea. 

How about writing a command line tool and hook it up through vscode tasks or a simple extension or other existing On-Save extensions first?

**Reply by rodrigocfd** [reply]
Indeed, sorting the methods should be an optional thing.

A VSCode extension would satisfy me needs at the moment, but I never wrote one. Do you recommend any resources where I can learn it?
